core/state/statedb.go

```diff
@@ -1268,58 +1268,63 @@ func (s *StateDB) Commit(deleteEmptyObjects bool) (common.Hash, error) {
 // - Add destination to access list
 // - Add precompiles to access list
 // - Add the contents of the optional tx access list
-func (s *StateDB) PrepareAccessList(sender common.Address, dst *common.Address, precompiles []common.Address, list types.AccessList) {
```

Comment:
Could you elaborate on what the debug boolean gives us?

Why is using the tracers outside of the engine not preferred?

Reply:
Looks like we are using the tracers and debug gives us the ability to actually populate the access list.

Reply:
The debug boolean allows us to simulate the EVM without it reverting due to no access list (when we are trying to construct an access list)